source/rpt/rpt_cell_reconstruction.cc

```diff
@@ -0,0 +1,626 @@
+                                 known_positions[i_particle][0],
+                               2) +
+                      std::pow(reconstruction_positions[i_particle][1] -
+                                 known_positions[i_particle][1],
+                               2) +
+                      std::pow(reconstruction_positions[i_particle][2] -
+                                 known_positions[i_particle][2],
+                               2));
+
+          // Associate status to a status id
+          unsigned int status_id;
+          if (cell_status[i_particle] == "right_cell")
+            status_id = 0;
+          else if (cell_status[i_particle] == "cost_function|right_cell")
+            status_id = 1;
+          else if (cell_status[i_particle] == "parent_cell|right_cell")
+            status_id = 2;
+          else if (cell_status[i_particle] ==
+                   "parent_cell|cost_function|right_cell")
+            status_id = 3;
+          else if (cell_status[i_particle] == "wrong_cell")
+            status_id = 4;
+          else if (cell_status[i_particle] == "cost_function|wrong_cell")
+            status_id = 5;
+          else if (cell_status[i_particle] == "parent_cell|wrong_cell")
+            status_id = 6;
+          else if (cell_status[i_particle] ==
+                   "parent_cell|cost_function|wrong_cell")
+            status_id = 7;
+
+
+          myfile << sep << cell_status[i_particle] << sep << status_id << sep
+                 << distance << std::endl;
+        }
+      else
+        myfile << std::endl;
+    }
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::assign_detector_positions()
+{
+  // Read text file with detector positions and store it in vector
+  std::ifstream detector_file(
+    parameters.detector_param.detector_positions_file);
+
+  std::string skip;
+  std::getline(detector_file, skip); // Skip header line
+  std::vector<double> values;
+  std::copy(std::istream_iterator<double>(detector_file),
+            std::istream_iterator<double>(),
+            std::back_inserter(values));
+
+  // Get the number of detector (2 positions for 1 detector, face and middle)
+  int number_of_detector = values.size() / (2 * dim);
+
+  // Extract positions, create point objects and detectors
+  for (int i = 0; i < number_of_detector; i++)
+    {
+      Point<dim> face_point(values[2 * dim * i],
+                            values[2 * dim * i + 1],
+                            values[2 * dim * i + 2]);
+      Point<dim> middle_point(values[2 * dim * i + dim],
+                              values[2 * dim * i + dim + 1],
+                              values[2 * dim * i + dim + 2]);
+
+      Detector<dim> detector(parameters.detector_param,
+                             i,
+                             face_point,
+                             middle_point);
+
+      detectors.push_back(detector);
+    }
+}
+
+template <int dim>
+void
+RPTCellReconstruction<dim>::read_known_positions()
+{
```

Comment:
Same thing, I would move this function to a different file, this way all solvers can use it :).